identity-service/sequelize/migrations/20220512172226-add_notification_json_b.js

```diff
@@ -0,0 +1,61 @@
+'use strict'
+
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    // This migration requires each enum add in it's own query, otherwise Sequelize complains
+    // that this can't be performed within a transaction
+    return queryInterface.sequelize.query(
+      `ALTER TYPE "enum_SolanaNotifications_type" ADD VALUE 'TipSend'`)
+      .then(() =>
+        queryInterface.sequelize.query(`ALTER TYPE "enum_SolanaNotifications_type" ADD VALUE 'TipReceive'`)
+      ).then(() =>
+        queryInterface.sequelize.query(`ALTER TYPE "enum_SolanaNotifications_type" ADD VALUE 'Reaction'`)
+      ).then(() =>
+        queryInterface.sequelize.query(`ALTER TYPE "enum_SolanaNotifications_type" ADD VALUE 'SupporterRankUp'`)
+      ).then(() =>
+        queryInterface.sequelize.query(`ALTER TYPE "enum_SolanaNotifications_type" ADD VALUE 'SupportingRankUp'`)
+      ).then(() => queryInterface.addColumn('SolanaNotifications', 'metadata', {
+        type: Sequelize.JSONB,
+        allowNull: true
+      })).then(() =>
+        queryInterface.sequelize.query(
+          `CREATE INDEX IF NOT EXISTS "solana_notifications_metadata_tip_tx_signature_idx" ON "SolanaNotifications"(("metadata"->>'tipTxSignature'))`
```

Comment:
Forgive me I know we've talked a lot about the data model/design here, but I'm trying to remember - why do we need an index here? Do we query notifications by tip ID? I thought we were going to go to DN for the reaction/tip and not do a self join in notifications to keep notifs/reactions uncoupled.

Reply:
Discussed offline